relay-server/src/services/processor/replay.rs

```diff
@@ -105,6 +109,13 @@ pub fn process(
         }
     }
 
+    // If the envelope contained a mobile event and mobile replay is disabled drop
+    // the envelope.
+    if contains_mobile_event && !mobile_replay_enabled {
+        state.managed_envelope.drop_items_silently();
+        return Ok(());
+    }
```

Comment:
We fully process the envelope before quitting, can we not quit out early if the feature flag is not enabled?

Reply:
Added early exit.